rust/stub/src/uefi_helpers.rs

```diff
@@ -1,12 +1,183 @@
 use core::ffi::c_void;
 
-use alloc::vec::Vec;
+use alloc::format;
+use alloc::string::ToString;
+use alloc::{vec, vec::Vec};
+use uefi::Guid;
 use uefi::{
-    prelude::BootServices,
-    proto::{loaded_image::LoadedImage, media::file::RegularFile},
-    Result,
+    prelude::{BootServices, RuntimeServices},
+    proto::{loaded_image::LoadedImage, media::file::RegularFile, device_path::text::{DevicePathToText, DisplayOnly, AllowShortcuts}},
+    Result, table::{runtime::{VariableVendor, VariableAttributes}, SystemTable, Boot}, CStr16, cstr16,
 };
 
+use crate::part_discovery::disk_get_part_uuid;
+use bitflags::bitflags;
+
+// systemd loader's GUID
+// != systemd's GUID
+// FIXME: please fix me, I hate UEFI.
+// https://github.com/systemd/systemd/blob/main/src/boot/efi/util.h#L114-L121
+const SD_LOADER: VariableVendor = VariableVendor(Guid::from_values(
+        0x4a67b082,
+        0x0a4c,
+        0x41cf,
+        u16::from_le_bytes([0xb6, 0xC7]),
+        u64::from_le_bytes([0xb6, 0xc7, 0x44, 0x0b, 0x29, 0xbb, 0x8c, 0x4f])
+    ));
```

Comment:
You can simplify this a bit with the `guid` macro:

```rust
pub const SD_LOADER: VariableVendor = VariableVendor(guid!("4a67b082-0a4c-41cf-b6c7-440b29bb8c4f"));
```

Reply:
Thank you!